src/index.ts

```diff
@@ -123,92 +153,100 @@ export async function downloadEntityAndContentFiles(
 }
 
 /**
- * Gets a stream of all the entities deployed to a server.
- * Includes all the entities that are already present in the server.
  * Accepts a fromTimestamp option to filter out previous deployments.
  *
  * @public
  */
-export async function* getDeployedEntitiesStream(
-  components: SnapshotsFetcherComponents,
-  options: DeployedEntityStreamOptions
-): AsyncIterable<DeploymentWithAuthChain> {
-  const logs = components.logs.getLogger(`getDeployedEntitiesStream(${options.contentServer})`)
-  // the minimum timestamp we are looking for
+export async function* getDeployedEntitiesStreamFromSnapshot(
+  components: SnapshotsFetcherComponents & {
+    processedSnapshots: IProcessedSnapshotsComponent
+  },
+  options: SnapshotDeployedEntityStreamOptions,
+  snapshotInfo: SnapshotInfo
+) {
+  const { greatestEndTimestamp, replacedSnapshotHashes, servers, snapshotHash } = snapshotInfo
   const genesisTimestamp = options.fromTimestamp || 0
+  const logs = components.logs.getLogger('getDeployedEntitiesStreamFromSnapshot')
+  logs.info('Snapshot to be processed.', { hash: snapshotHash, contentServers: JSON.stringify(Array.from(servers)) })
+  const shouldStreamSnapshot =
+    greatestEndTimestamp > genesisTimestamp &&
+    await components.processedSnapshots.shouldProcessSnapshot(snapshotHash, replacedSnapshotHashes)
 
-  // the greatest timestamp we processed
-  let greatestProcessedTimestamp = genesisTimestamp
-
-  const metricLabels = contentServerMetricLabels(options.contentServer)
-
-  // 1. get the hash of the latest snapshot in the remote server, retry 10 times
-  const { hash, lastIncludedDeploymentTimestamp } = await getGlobalSnapshot(
-    components,
-    options.contentServer,
-    options.requestMaxRetries
-  )
-
-  // 2. download the snapshot file if it contains deployments
-  //    in the range we are interested (>= genesisTimestamp)
-  if (hash && lastIncludedDeploymentTimestamp && lastIncludedDeploymentTimestamp > genesisTimestamp) {
+  if (shouldStreamSnapshot) {
     try {
       // 2.1. download the snapshot file if needed
```

Comment:
check the comments here, as there is no `// 1.` now

Reply:
updated